tasks/sep/unichain-003-petra-l1-upgrade-new-prestate/SignFromJson.s.sol

```diff
@@ -0,0 +1,529 @@
+    FaultDisputeGame faultDisputeGame;
+    PermissionedDisputeGame permissionedDisputeGame;
+    address newMips;
+    address oracle;
+    uint256 chainId;
+
+    address[] extraStorageAccessAddresses;
+
+    constructor() SuperchainRegistry("sepolia", "unichain", "v1.8.0-rc.4") {}
+
+    function setUp() public {
+        dgfProxy = DisputeGameFactory(proxies.DisputeGameFactory);
+        newMips = standardVersions.MIPS.Address;
+        oracle = standardVersions.PreimageOracle.Address;
+        chainId = chainConfig.chainId;
+
+        string memory inputJson;
+        string
+            memory path = "/tasks/sep/unichain-003-petra-l1-upgrade-new-prestate/input.json";
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (
+            string memory data
+        ) {
+            inputJson = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+
+        faultDisputeGame = FaultDisputeGame(
+            stdJson.readAddress(
+                inputJson,
+                "$.transactions[0].contractInputsValues._impl"
+            )
+        );
+        permissionedDisputeGame = PermissionedDisputeGame(
+            stdJson.readAddress(
+                inputJson,
+                "$.transactions[1].contractInputsValues._impl"
+            )
+        );
+
+        _precheckDisputeGameImplementation(
+            GameType.wrap(0),
+            address(faultDisputeGame)
+        );
+        _precheckDisputeGameImplementation(
+            GameType.wrap(1),
+            address(permissionedDisputeGame)
+        );
+        // INSERT NEW PRE CHECKS HERE
+    }
+
+    function getCodeExceptions()
+        internal
+        view
+        override
+        returns (address[] memory)
+    {
+        return new address[](0);
+    }
+
+    // _precheckDisputeGameImplementation checks that the new game being set has the same configuration as the existing
+    // implementation with the exception of the absolutePrestate. This is the most common scenario where the game
+    // implementation is upgraded to provide an updated fault proof program that supports an upcoming hard fork.
+    function _precheckDisputeGameImplementation(
+        GameType _targetGameType,
+        address _newImpl
+    ) internal view {
+        console.log(
+            "pre-check new game implementations",
+            _targetGameType.raw()
+        );
+
+        FaultDisputeGame currentImpl = FaultDisputeGame(
+            address(dgfProxy.gameImpls(GameType(_targetGameType)))
+        );
+        // No checks are performed if there is no prior implementation.
+        // When deploying the first implementation, it is recommended to implement custom checks.
+
+        FaultDisputeGame faultDisputeGame = FaultDisputeGame(_newImpl);
+        // these are both using the latest version of the MIPs contracts
+        // require(address(currentImpl.vm()) != address(faultDisputeGame.vm()), "10");
+        require(
```

Comment:
Worth adding a check to require `faultDisputeGame.gameType() == _targetGameType`.